src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/PublicKey.cs

```diff
@@ -49,5 +71,69 @@ public AsymmetricAlgorithm Key
         }
 
         public Oid Oid => _oid;
+
+        public bool TryExportSubjectPublicKeyInfo(Span<byte> destination, out int bytesWritten) =>
+            EncodeSubjectPublicKeyInfo().TryEncode(destination, out bytesWritten);
+
+        public byte[] ExportSubjectPublicKeyInfo() =>
+            EncodeSubjectPublicKeyInfo().Encode();
+
+        public static PublicKey CreateFromSubjectPublicKeyInfo(ReadOnlySpan<byte> source, out int bytesRead)
+        {
+            int read = DecodeSubjectPublicKeyInfo(
+                source,
+                out Oid localOid,
+                out AsnEncodedData localParameters,
+                out AsnEncodedData localKeyValue);
+
+            bytesRead = read;
+            return new PublicKey(localOid, localParameters, localKeyValue);
+        }
+
+        private AsnWriter EncodeSubjectPublicKeyInfo()
+        {
+            SubjectPublicKeyInfoAsn spki = new SubjectPublicKeyInfoAsn {
+                Algorithm = new AlgorithmIdentifierAsn {
+                    Algorithm = _oid.Value ?? string.Empty,
+                    Parameters = EncodedParameters.RawData
+                },
+                SubjectPublicKey = EncodedKeyValue.RawData
+            };
+
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            spki.Encode(writer);
+            return writer;
+        }
+
+        private static unsafe int DecodeSubjectPublicKeyInfo(
+            ReadOnlySpan<byte> source,
+            out Oid oid,
+            out AsnEncodedData parameters,
+            out AsnEncodedData keyValue)
+        {
+            fixed (byte* ptr = &MemoryMarshal.GetReference(source))
+            using (MemoryManager<byte> manager = new PointerMemoryManager<byte>(ptr, source.Length))
+            {
+                AsnValueReader reader = new AsnValueReader(source, AsnEncodingRules.DER);
+
+                int read;
+                SubjectPublicKeyInfoAsn spki;
+
+                try
+                {
+                    read = reader.PeekEncodedValue().Length;
+                    SubjectPublicKeyInfoAsn.Decode(ref reader, manager.Memory, out spki);
+                }
+                catch (AsnContentException e)
+                {
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding, e);
+                }
+
+                oid = new Oid(spki.Algorithm.Algorithm);
```

Comment:
```suggestion
                oid = new Oid(spki.Algorithm.Algorithm, null);
```

To avoid the lookup to see if it was an OID value or a FriendlyName